### ISSUE TITLE
Grid (12750)

### ISSUE THREAD

**Issue by Cabanur** [ValveSoftware/Proton#1061]
While being playing, I recieved a steam message. Initially, I could alt-tab normally to respond and go back to the game. After going back to the steam chat window and then back to the game, it had a tremendous loss of performance (to around 5-10 fps from a very stable 60+) becoming unplayable. 

I alt-tabbed again to go back to chat and check on my system monitor. I observed the whole system was slow and sluggish to respond, to the point that firefox crashed and the steam chat window was unresponsive. At this point the game crashed and closed itself. Here's the [proton log](https://github.com/ValveSoftware/Proton/files/2339823/steam-12750.log) The original file was 40MB so I just cut around the 100k line until close to the end, seeing as it is mostly repeated stuff. I am keeping the original file and can upload it if useful.

Even after the game crashed and as I write this, the steam chat has an odd behaviour. The friend list window has cloned itself, only one of them responding to clicks. 

[System info](https://gist.github.com/Cabanur/be3f08865bdd424ca28a309e2ecfbf61): Ryzen 1600x, Radeon RX 580 on Debian Testing.

**Comment by kisak-valve** [comment]
Grid (12750) Crashes when launching a race

Issue transferred from https://github.com/ValveSoftware/Proton/issues/1160.
@hswhite33 posted on 2018-09-02T01:54:22:

Managed to get into the menus after a bit of fiddling (I had to do the same fiddling in Windows though, NBD). Game crashes consistently when trying to launch a race, I haven't managed to successfully launch one.

[steam-12750.log](https://github.com/ValveSoftware/Proton/files/2342909/steam-12750.log)

[System info](https://gist.github.com/hswhite33/d5872273b13aa91e29a7104c1c4866a9)

Proton 3.7-5 Beta

**Comment by sputniktr02** [comment]
I can confirm the issue. I managed to get into race couple of times but game crashes a couple of seconds into the race or when i change the camera view, in car view had some visual artifacts that i could see briefly before game crashed.
Log: 
[steam-12750.log](https://github.com/ValveSoftware/Proton/files/2343692/steam-12750.log)
 
System Info: https://gist.github.com/sputniktr02/3f410eb3d6206ab7bd35b70c6a1bb12c

Proton 3.7-5 Beta

**Comment by sputniktr02** [comment]
Tested with Proton 3.7-3. I was able to finish a race on Istanbul Park but as soon as race finished game crashed. Tested again on same track with same car, this time game did not crash but some interface textures was missing while game was displaying race results. Selected different car on the same track and game crashed again. Picked another track and game crashed upon launching the race. I think issue might be related to shader cashing or something. Because when you select same car and track after each crash you can go further each time and eventually finish a race.

**Comment by Leopard1907** [comment]
Proton 3.7.5 beta. Game starts and crashes within 10 seconds.

Linux Mint 18.3

GTX 1050 - 396.54 driver

**Comment by Leopard1907** [comment]
Here is the log of it.

[steam-12750.log](https://github.com/ValveSoftware/Proton/files/2363812/steam-12750.log)

**Comment by EidoJ** [comment]
Hi Guys. Since installing it on Steam a few weeks ago it's been crashing every few races for me. Now it's at my second 24h of Le Mans and I can't load the race at all. The loading screen completes and I hear 1/2 a second of sound (screen still black) before it crashes.

Specs:
CPU: FX-8350
GPU: R9 290 mesa: OpenGL version string: 4.5 (Compatibility Profile) Mesa 19.0.0-devel (git-d28bc35 2018-11-08 xenial-oibaf-ppa)
Kernel: UKUU updated it to V4.18.13
OS: elementary OS 0.4.1

Crash log: [steam-12750.log](https://github.com/ValveSoftware/Proton/files/2641289/steam-12750.log)

**Comment by kisak-valve** [comment]
```
radeon: mmap failed, errno: 12
854.148:0008:0048:err:d3d:wined3d_debug_callback 0x21b268: "GL_OUT_OF_MEMORY in glMapBufferRange(map failed)".
radeon: mmap failed, errno: 12
854.148:0008:0048:err:d3d:wined3d_debug_callback 0x21b268: "GL_OUT_OF_MEMORY in glMapBufferRange(map failed)".
```
This looks like the lines of interest in that log.

**Comment by gaab0r** [comment]
I have the same problem, crushing after race starts.
I turned of steam overlay for the game and now it doesn't crush but my controller doesn't work.

**Comment by Leopard1907** [comment]
It works now on Proton 3.16-6 Beta. It was crashing after first race until i put that command into launch options. Now it fully works.

`PROTON_FORCE_LARGE_ADDRESS_AWARE=1 %command%`

For reference:

https://github.com/ValveSoftware/Proton/wiki/Changelog#316-6

**Comment by Leopard1907** [comment]
Yup , it is working without problems. I think that game is suitable for whitelisting with Large Adress Aware variable.

@kisak-valve 

My system info:

https://gist.github.com/Leopard1907/e1fd929da6d76c1938ce5c4f2a036101

**Comment by kisak-valve** [comment]
With Proton 3.16-6 and `PROTON_FORCE_LARGE_ADDRESS_AWARE=1 %command%` in the game's launch options, is anyone having any issues with this game that would block it from being whitelisted?

**Comment by sputniktr02** [comment]
@kisak-valve 

Game is working for me as well, performance is very good. If i have time i'll also test with driving wheel (Logitech Driving Force GT) and report back, probably tomorrow.

System Info: https://gist.github.com/sputniktr02/3f410eb3d6206ab7bd35b70c6a1bb12c

**Comment by leillo1975** [comment]
The game works perfect with PROTON_FORCE_LARGE_ADDRESS_AWARE=1 %command% , my Steam Controller is recognized and works well, but I can't drive with my Logitech G29. I can't set axis and buttons on options. I don't know if this wheel is recognized in Windows. In another term, The maximum resolution for me is 1280x720, I can't set 1920x1080

**Comment by kisak-valve** [comment]
http://forums.codemasters.com/discussion/38662/logitech-g29-cant-use-wheel-with-grid2-grid-autosport seems to indicate the game does not support the G29 out of the box. Maybe give https://steamcommunity.com/app/255220/discussions/0/371918937269501950/ a try? This is under the assumption that older games do not have more support than newer games.

**Comment by leillo1975** [comment]
Take a note, that you are talking about GRID Autosport (255220, with native support), not GRID (the first):
https://steamcommunity.com/sharedfiles/filedetails/?id=848110469

About 1920x1080 resolution this can be fixed editing one file:
https://steamcommunity.com/app/12750/discussions/0/152392549361254851/

**Comment by sputniktr02** [comment]
@kisak-valve 
Hi, i have tested the game with my Driving Force GT and it is working properly with force feedback. I just had to bind the controls manually.

**Comment by leillo1975** [comment]
> @kisak-valve
> Hi, i have tested the game with my Driving Force GT and it is working properly with force feedback. I just had to bind the controls manually.

I can't do that with G29

**Comment by hswhite33** [comment]
My G29 doesn't work with this game in Windows either; I don't believe this is a Proton issue.

Confirming that this game works identically to Windows in Proton 3.16-6 with `PROTON_FORCE_LARGE_ADDRESS_AWARE=1` set:

* 1080p works after modifying the `hardware_settings_restrictions.xml` file (this step is required in Windows as well)
* Runs above 60fps at max settings at 1080p
* No crashes observed

[System info](https://gist.github.com/hswhite33/d7aa5164dfdc90e8083aed310cb04aef)

**Comment by Leopard1907** [comment]
@EidoJ @Cabanur

Can you retest the game on Proton 3.16-6 with PROTON_FORCE_LARGE_ADDRESS_AWARE=1 

Since it is working for us with it, maybe it will also work for you with that way.

**Comment by EidoJ** [comment]
@Leopard1907 I'll do so in a few days. My gaming rig is packed away while I move.

**Comment by EidoJ** [comment]
So that didn't fix it for me, although it actually let me get into the race this time.
The camera got up to the guy they tell you to watch out for in the race, and then the game exited as before. Any clues why that might help? I'm a firmware guy, so I don't know too much about why large addresses will help with a 32-bit game. Was that a standard feature of Windows that isn't present by default on Wine/Proton?

**Comment by Leopard1907** [comment]
@EidoJ 
Are you sure you're on Proton 3.16-6 Beta?

Large Adress Aware introduced with that version.

I don't know the details.

**Comment by EidoJ** [comment]
> @EidoJ
> Are you sure you're on Proton 3.16-6 Beta?
> 
> Large Adress Aware introduced with that version.
> 
> I don't know the details.

You guys are absolutely right, I was being a bit of an idiot. 3.16-6 Beta works just fine with that force large address aware thing. Thanks!

**Comment by Leopard1907** [comment]
No problem , have fun.

**Comment by Leopard1907** [comment]
There is a regression happened on my end with 3.16-8 Beta. Game kills itself after first Codemasters screen.

[steam-12750.log](https://github.com/ValveSoftware/Proton/files/2974552/steam-12750.log)


I think there is some regression specifially hits my end ( which is contains Turkish locale ) , regressions occurred on also Skyrim SE and Witcher 1.

@aeikum 

 I also tried with workaround you suggested on Witcher 1 issue but no luck.

**Comment by kisak-valve** [comment]
Hello @Leopard1907, what was the last known working version of Proton for you? 3.16-6?

**Comment by Leopard1907** [comment]
Yes , it was 3.16-6. Sadly i didn't try with 3.16-7 in that time

**Comment by aeikum** [comment]
I set up my machine so that Witcher 1 crashes when running in 3.16-8, but works in 4.2-2.  Grid works fine in both 3.16-8 and 4.2-2. In other words, I can't reproduce. You see this crash immediately after the intro videos, before the main menu?

**Comment by Leopard1907** [comment]
Yes , it crashes at second intro video ( where Codemasters name written on road ).

And also ; intro videos doesn't have sound. That first intro video with engine revs up is dead silent.

Here is the 4.2-2 log.

[steam-12750.log](https://github.com/ValveSoftware/Proton/files/3039663/steam-12750.log)

**Comment by aeikum** [comment]
This looks bad:
```
3424.049:0027:0039:trace:module:GetModuleFileNameW L"Z:\\home\\asus\\.local\\share\\Steam\\steamapps\\common\\Grid\\grid.exe"
mmap() failed: Bellek ayrılamadı
Failed to create permanent mapping for memfd region with ID = 126527006
Ignoring received block reference with non-registered memfd ID = 126527006
3424.049:0027:0039:err:dsound:DSOUND_ReopenDevice Initialize failed: 8889000f
```
Something funky with your PulseAudio setup? Nothing like that appeared in your previous log, though.

Can you get another log with +relay added to the logging channels? You'll want to set up and modify user_settings.py for Proton 4.2 (and you'll want to remove those changes when you are done gathering a log). Let me know if you need more instruction than that. The relay log will be very large, so you'll probably need to compress it with `xz` before you upload it here.

**Comment by Leopard1907** [comment]
I guess nothing is funky , since other games ( native + Proton ) have sounds.

I edited sample.py as user.settings.py and added +relay to winedebug line. I guess i was successful since it generated a 430 mb log.

Here is the log , compressed as `xz`

Pre edit: Github doesn't allow xz so here is gz.

[steam-12750.log.gz](https://github.com/ValveSoftware/Proton/files/3039772/steam-12750.log.gz)

**Comment by Leopard1907** [comment]
@aeikum 

At least sound issue got solved. Credits goes to @Tk-Glitch

Basically , an OpenAL issue.

I had to go into `*/compatdata/12750/pfx/drive_c/windows/syswow64/` and grab OpenAL32.dll from there and copy it into game directory.

Then with using winecfg `WINEPREFIX=~/.steam/steam/steamapps/compatdata/12750/pfx/ winecfg` i set openal as an override as `native` .

Game still crashes though , despite i use Large Adress Aware and set win version to XP through winecfg.

**Comment by lucifertdark** [comment]
Tested with Proton 4.2-3 I had to kill -9 the executable as it crashed on exit, apart from that it worked perfectly.

**Comment by Leopard1907** [comment]
@aeikum @kisak-valve I can confirm regression on my end is fixed with 4.2-6.

**Comment by aeikum** [comment]
Great, thanks for reporting and retesting. By the way, our QA found that setting the Large Address Aware flag (see the proton README) helps prevent some crashes. That might be helpful for you.

**Comment by Leopard1907** [comment]
Yes, i'm aware of that. I asked about for a retest to users on 3.16-6 with Large Adress Aware but later down the road 3.16-8 regression happened. :)

https://github.com/ValveSoftware/Proton/issues/1061#issuecomment-454475099

**Comment by aeikum** [comment]
Ah, I missed that conversation on this bug! Maybe they got the idea to try it from you :)

**Comment by spacegears** [comment]
Sooo.... over the last couple of weeks, I was playing GRID (2008) on Proton 7 without any issues. Now that Proton 8 is out, it seems to crash on launch. After reverting to 7.0-6, the game launches again without any issues.

Here is the Proton 8.0-1 log:
[steam-12750.log](https://github.com/ValveSoftware/Proton/files/11296751/steam-12750.log)

**Comment by kisak-valve** [comment]
Hello @spacegears, `err:module:import_dll Library OpenAL32.dll (which is needed by L"Z:\\mnt\\Games\\Steam\\steamapps\\common\\Grid\\grid.exe") not found` looks like the line of interest from your log.

**Comment by AJuujarvi** [comment]
@spacegears Hello! Can you try again after deleting the prefix "12750" found in ~/.steam/steam/steamapps/compatdat please? I just took a look at this and that seemed to fix the issue for me. I was able to get it working on Proton 8.0-1d as well as experimental-8.0-20230417.

**Comment by spacegears** [comment]
Whoa, thanks for looking into it, I can confirm that now it works on both Proton 8.0-1 as well as Proton Experimental (8.0-20230417).

~~What's strange though is that I cannot seem to use my old saves anymore, but I'm not 100% sure that's not just me screwing it up. I'm still looking into it.~~ Update: Yeah, it was me. Everything works.

Thanks again @AJuujarvi and @kisak-valve!
